### PR TITLE
Add Backend Exception Handling for Debts

### PR DESCRIPTION
**Pull request by camerontesterman**
## Description of change
* Adds exception handling to the Debts service. 
* Adds a 400 Bad Request exception for debts within the exceptions yml file.
* Adds JSON schema validation for Debts responses.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/11791

## Things to know about this PR
Tested and verified locally.

### CODE REVIEW

File: config/locales/exceptions.en.yml [hunk 1]
```diff
@@ -384,6 +384,11 @@ en:
         code: 'CASEFLOWSTATUS502'
         detail: 'Received an invalid response from the upstream server'
         status: 502
+      DEBTS400:
+        <<: *external_defaults
+        title: 'Bad Request'
+        code: 'DEBTS400'
+        detail: 'Received a bad request response from the upstream server'
```
Comment (f1337) [inline config/locales/exceptions.en.yml:391]: Should we also include `status: 400`?

File: lib/debts/service.rb [hunk 2]
```diff
@@ -2,10 +2,57 @@
...
+        yield
+      end
+    rescue => e
+      handle_error(e)
+    end
+
+    def save_error_details(error)
+      Raven.tags_context(
+        external_service: self.class.to_s.underscore
+      )
```
Comment (f1337) [inline lib/debts/service.rb:30]: Do we want to include tests to ensure we're sending the expected details to Raven?

File: config/locales/exceptions.en.yml (hunk 1 above)
Comment (camerontesterman) [reply config/locales/exceptions.en.yml:391]: good catch! added.

File: lib/debts/service.rb (hunk 2 above)
Comment (camerontesterman) [reply lib/debts/service.rb:30]: yeah that's a good idea. I'll add that and then resolve this convo.

Comment (camerontesterman) [reply lib/debts/service.rb:30]: Added

**Pull request merged**